server/src/main/java/org/spine3/server/reflect/CommandHandlerMethod.java

```diff
@@ -67,11 +116,99 @@ public CommandHandlerMethod(Method method) {
     @CheckReturnValue
     public static Set<CommandClass> getCommandClasses(Class<?> cls) {
         final Set<CommandClass> result = CommandClass.setOf(
-                getHandledMessageClasses(cls, PREDICATE));
+                getHandledMessageClasses(cls, predicate()));
         return result;
     }
 
     /**
+     * Ensures that the passed instance of {@code Message} is not an {@code Any},
+     * and unwraps the command message if {@code Any} is passed.
+     */
+    private static Message ensureCommandMessage(Message msgOrAny) {
+        Message commandMessage;
+        if (msgOrAny instanceof Any) {
+            /* It looks that we're getting the result of `command.getMessage()`
+               because the calling code did not bother to unwrap it.
+               Extract the wrapped message (instead of treating this as an error).
+               There may be many occasions of such a call especially from the
+               testing code. */
+            final Any any = (Any) msgOrAny;
+            commandMessage = AnyPacker.unpack(any);
+        } else {
+            commandMessage = msgOrAny;
+        }
+        return commandMessage;
+    }
+
+    /**
+     * Invokes the handler method in the passed object.
+     *
+     * @return the list of events produced by the handler method
+     */
+    public static List<? extends Message> invokeHandler(Object object,
+                                                        Message command,
+                                                        CommandContext context) {
+        checkNotNull(command);
+        checkNotNull(context);
+        final Message commandMessage = ensureCommandMessage(command);
+
+        try {
+            final CommandHandlerMethod method = forMessage(object.getClass(),
+                                                           commandMessage);
+            final List<? extends Message> eventMessages = method.invoke(object,
+                                                                        commandMessage,
+                                                                        context);
+            return eventMessages;
+        } catch (InvocationTargetException e) {
+            throw wrappedCause(e);
+        }
+    }
+
+    /**
+     * Creates new {@code CommandContext} with passed parameters.
+     *
+     * @param producerId the ID of an object which produced the event
+     * @param version optional version of the object
+     * @param commandContext the context of the command handling of which produced the event
+     * @return new {@code CommandContext}
+     */
+    public static EventContext createEventContext(Any producerId,
```

Comment:
As long as it is `public`, we should ensure both `producerId` and `commandContext` aren't `null`s.

I also assume we don't have a nullability check for the `static` methods of this class in tests.